### PR TITLE
added test_listfile

### PR DESCRIPTION
**Pull request by Kabilan-16**
Added  a new test file for testing a listfile api

### CODE REVIEW

File: src/backend/chat_service/chat_service.py [hunk 1]
```diff
@@ -133,6 +133,10 @@ async def upload_File(request: Request, customer_guid: str = Form(...), file:Upl
 async def list_files(request: Request, customer_guid: str):
     logger.debug(f"Entering list_files() with Correlation ID: {request.state.correlation_id}")
     try:
+        if not db_manager.check_customer_guid_exists(customer_guid):
```
Comment (SaravanakumarR2018) [inline src/backend/chat_service/chat_service.py:136]: Instead of checking whether customer_guid exists here, you can check whether customer_guid exists in minio_manager by verifying whether customer_guid bucket exists.

If it does not exist you can throw an exception saying "customer_guid" not found with the correct HTTP ERROR CODE

Remove these lines
        if not db_manager.check_customer_guid_exists(customer_guid):
            logger.error(f"Invalid customer_guid: {customer_guid}")
            raise HTTPException(status_code=404, detail="Invalid customer_guid provided")

File: test/IntegrationTests/test_listfile.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        if response.status_code==200:
+            logger.info("Response status code is valid and matches the expected value (200). File listing successful.")
+        else:
+            logger.error(f"Unexpected status code: {response.status_code}. Expected: 200.")
+            raise AssertionError(f"Expected status code 200 but got {response.status_code}")
+
+        # Assert the response
+        self.assertEqual(response.status_code, 200, "Failed to list files for valid customer_guid")
+        self.assertIn("files", response.json(), "No 'files' key returned in response")
+
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_listfile.py:54]: Verify the names of the file here

Try creating a separate customer_guid and upload the files local to the function instead of setup function

File: test/IntegrationTests/test_uploadfile.py
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_uploadfile.py:1]: Remove this file

File: test/IntegrationTests/test_listfile.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s'
+)
+logger=logging.getLogger(__name__)
+
+class TestListFileAPI(unittest.TestCase):
+    BASE_URL="http://localhost:8000"
+
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_listfile.py:13]: Add testcases like
1) Adding two files and verifying whether both the file names are listed
2) Adding the third file and verifying whether all three filenames are listed

File: src/backend/chat_service/chat_service.py [hunk 4]
```diff
@@ -133,6 +133,10 @@ async def upload_File(request: Request, customer_guid: str = Form(...), file:Upl
 async def list_files(request: Request, customer_guid: str):
     logger.debug(f"Entering list_files() with Correlation ID: {request.state.correlation_id}")
     try:
+        if not minio_manager.client.bucket_exists(customer_guid):
```
Comment (SaravanakumarR2018) [inline src/backend/chat_service/chat_service.py:136]: chat_service.list_files should only call minio_manager.list_files. All checks, including bucket existence validation, should be handled within minio_manager.list_files.

If the bucket doesn't exist, ensure that an appropriate error is returned. This will help raise the correct error code and message from chat_Service.list_files.

File: test/IntegrationTests/test_listfile.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+        #Verify the response contains the 'detail' field
+        self.assertIn("detail", data, "'detail' not found in response data")
+        logger.info("'detail' found in response data, verifying its content")
+
+        #Verify that the error message indicates invalid customer_guid
+        self.assertEqual(data["detail"], "Invalid customer_guid provided", "Unexpected error message content")
+
+        logger.info("Test completed successfully for test_list_files_from_invalid_customer_guid")
+
+    def test_list_files_two_files_uploaded(self):
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_listfile.py:178]: Do we think that these two testcases are the same functionality wise?

 def test_list_files_from_valid_customer(self):
 def test_list_files_two_files_uploaded(self):
 
 If yes should we remove one of them?

File: test/IntegrationTests/test_listfile.py [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s'
+)
+logger=logging.getLogger(__name__)
+
+class TestListFileAPI(unittest.TestCase):
+    BASE_URL="http://localhost:8000"
+
+    def setUp(self):
```
[the remark this answers is not in the document]
Comment (Kabilan-16) [reply test/IntegrationTests/test_listfile.py:14]: It's just add a customer_guid

File: src/backend/chat_service/chat_service.py (hunk 4 above)
Comment (Kabilan-16) [reply src/backend/chat_service/chat_service.py:136]: Okey sir

File: src/backend/chat_service/chat_service.py [hunk 7]
```diff
@@ -141,13 +141,16 @@ async def list_files(request: Request, customer_guid: str):
             logger.info(f"No files found in bucket '{customer_guid}'")
 
         return {"files": file_list}
-    except Exception as e:
```
Comment (SaravanakumarR2018) [inline src/backend/chat_service/chat_service.py:144]: YOu should always capture Base classException instead of a specific exception

Post capturing the exception you can use if else to specify the reason for exception

File: src/backend/chat_service/chat_service.py [hunk 8]
```diff
@@ -141,13 +141,16 @@ async def list_files(request: Request, customer_guid: str):
...
-        logger.error(f"Error listing files:'{customer_guid}': {e}")
-        raise HTTPException(status_code=500, detail="Error listing files")
+    except HTTPException as e:
+        if e.status_code==404:
+            logger.error(f"Bucket not found for customer_guid '{customer_guid}': {e.detail}")
```
Comment (SaravanakumarR2018) [inline src/backend/chat_service/chat_service.py:146]: Change this to a common log instead of being specific "Bucket not found for customer_guid"

File: src/backend/minio/minio_manager.py [hunk 9]
```diff
@@ -19,6 +20,7 @@ def __init__(self):
                 secret_key=os.getenv('MINIO_ROOT_PASSWORD','minioadmin'),
                 secure=False
             )
+
```
Comment (SaravanakumarR2018) [inline src/backend/minio/minio_manager.py:23]: Remove whitespace changes

File: src/backend/minio/minio_manager.py [hunk 10]
```diff
@@ -63,12 +65,12 @@ def upload_file(self,bucket_name,filename,file_data):
             logger.error(f"Unexpected error during file upload:{e}")
             return {"error":f"An error occurred:{e}"}
 
-    #list the files in the MinIO bucket
+    # list the files in the MinIO bucket
```
Comment (SaravanakumarR2018) [inline src/backend/minio/minio_manager.py:68]: Is this change needed?

File: src/backend/chat_service/chat_service.py [hunk 11]
```diff
@@ -255,4 +258,4 @@ async def check_server_status(request: Request):
 if __name__ == "__main__":
     import uvicorn
 
-    uvicorn.run(app, host="0.0.0.0", port=8000)
```
Comment (SaravanakumarR2018) [inline src/backend/chat_service/chat_service.py:258]: Why this change - make sure this line does not change

**Review by SaravanakumarR2018 — COMMENTED** [review]
DO not add new lines or change lines unnecessarily in pull request.
it will create unnecessary noise

**Review by SaravanakumarR2018 — APPROVED** [review]
